official-antistasi-community/A3-Antistasi
# ⚒ Refactored BattleMenu to Fully Qualified Classes

## What type of PR is this.
* Change
* Enhancement

### What have you changed and why?
Refactored the classes defined inside `defines.hpp` to have the A3A_core prefix.
This prevents it from overriding base game UIs and base game UIs overriding it.

### Please specify which Issue this PR Resolves.
closes [#2237](https://github.com/official-antistasi-community/A3-Antistasi/issues/2237)

### Please verify the following and ensure all checks are completed.
* Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
* Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 
1. Start Game.
2. Mouse over base game buttons and UI in server lobby. They should not turn red.
3. Open the battle menu (Y), and see that the font is Roboto, they are big red buttons. 

### Notes:
All changes can be reproduced with the following:
```
File Match:
A3A\addons\core\d*

Find & Replace:
Match Case & Whole Word & Regex

RscText(?!\\)
A3A_core_BattleMenuText

RscPicture(?!\\)
A3A_core_BattleMenuPicture

RscButton(?!\\)
A3A_core_BattleMenuRedButton

RscFrame(?!\\)
A3A_core_BattleMenuFrame

BOX(?!\\)
A3A_core_BattleMenuBOX

RscListBox(?!\\)
A3A_core_BattleMenuListBox
```
